src/App.tsx

```diff
@@ -1,31 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const options = [3, 5, 10, 20];
+  const [perPage, setPerPage] = useState(5);
+  const [pageNumber, setPageNumber] = useState(1);
+  const lastIndex = perPage * pageNumber;
+  const firstIndex = lastIndex - perPage;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${pageNumber} (items ${firstIndex + 1} - ${lastIndex > items.length ? items.length : lastIndex} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            value={perPage}
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            onChange={event => {
+              setPerPage(+event.target.value);
+              setPageNumber(1);
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here
check other places